### PR TITLE
Allow `_gce_` network when not using discovery gce

### LINKED ISSUE #15724: Allow `_gce_` network when not using discovery gce
For now we support `_gce_` only if discovery is set to `gce` and all information about GCE is provided (project_id and zone).
But in some cases, people would like to only bind to `_gce_` on a single node (without any elasticsearch cluster).

They could access the machine then from other machines running inside the same project.

From this thread: https://discuss.elastic.co/t/failed-to-start-elasticsearch-with-gce-cloud-plugin/38176/

### PR DESCRIPTION
**Pull request by dadoonet**
For now we support `_gce_` only if discovery is set to `gce` and all information about GCE is provided (project_id and zone).
But in some cases, people would like to only bind to `_gce_` on a single node (without any elasticsearch cluster).

They could access the machine then from other machines running inside the same project.

This commit adds a new GceMetadataService which is started as soon as the plugin is started so GceNameResolver can use it to resolve `_gce`.

Closes #15724.

### CODE REVIEW

**Comment by dadoonet** [comment]
@imotov Could you review it please? 
Note that I did that against 2.x branch. If merged, I'll port it to master branch.

File: plugins/cloud-gce/src/test/java/org/elasticsearch/discovery/gce/GceComputeServiceMock.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 import com.google.api.client.testing.http.MockLowLevelHttpRequest;
 import com.google.api.client.testing.http.MockLowLevelHttpResponse;
 import org.elasticsearch.cloud.gce.GceComputeServiceImpl;
+import org.elasticsearch.cloud.gce.GceMetadataServiceImpl;
```
Comment (imotov) [inline plugins/cloud-gce/src/test/java/org/elasticsearch/discovery/gce/GceComputeServiceMock.java]: Most of the imports here are not used

**Comment by imotov** [comment]
I think we typically open PRs against master and then backport to 2.0, not other way around. Left a cosmetic comment. LGTM, but it might make sense for someone who is more familiar with GCE to take another look.

**Comment by dadoonet** [comment]
Thanks @imotov.

@bleskes It's all yours now! :) Same as #15765 but on 2.x branch.

**Comment by dadoonet** [comment]
If no one objects, I'll probably close this PR as it will be most likely merged in master.
Not sure if we want to have that "feature" in 2.4?

**Comment by dadoonet** [comment]
As I mentioned a month ago, I'm closing this one.
It will be merged in master with https://github.com/elastic/elasticsearch/pull/15765

**Pull request closed without merging**